src/NovaAdapterBase.sol

```diff
@@ -12,44 +12,41 @@ abstract contract NovaAdapterBase is ERC20 {
     );
 
     address immutable sDAI;
-    ERC20 asset;
+    address immutable asset;
 
     constructor(
-        ERC20 _asset,
-        address _sDAI,
-        string memory _name,
-        string memory _symbol,
-        uint8 _decimals
-    ) ERC20(_name, _symbol, _decimals) {
+        address _asset,
+        address _sDAI
+    ) {
         asset = _asset;
         sDAI = _sDAI;
     }
 
     function deposit(uint256 assets) external returns (bool, uint256) {
-        bool success = asset.transferFrom(msg.sender, address(this), assets);
-        if(!success){
-            revert TransferFailed(msg.sender, address(this), assets);
-        }
+        bool successFirst = IERC20(asset).transferFrom(msg.sender, address(this), assets);
 
         (, int256 sDai) = _swap(int256(assets), true);
```

Comment:
The name sDai is already used for the sDai address variable, it's a little bit confusing here